File: CourseManagementSystem.API/ClientApp/src/app/viewmodels/testSubmissionWithUserInfoVM.ts
/**
 * class representing a test submission with information about an user
 */
export class TestSubmissionWithUserInfoVM {
  /**
   * email of the student
   */
  public studentEmail: string;

  /**
   * identifier of the test submission
   */
  public testSubmissionId: number;

  /**
   * percentual score from the test (0=0%,1=100%)
   */
  public percentualScore: number;

  constructor(studentEmail: string, testSubmissionId: number, percentualScore: number) {
    this.studentEmail = studentEmail;
    this.testSubmissionId = testSubmissionId;
    this.percentualScore = percentualScore;
  }
}

File: CourseManagementSystem.API/ViewModels/TestSubmissionWithUserInfoVM.cs
namespace CourseManagementSystem.API.ViewModels
{
    /// <summary>
    /// class representing info about test submission and course memeber
    /// </summary>
    public class TestSubmissionWithUserInfoVM
    {
        public TestSubmissionWithUserInfoVM() { }

        public TestSubmissionWithUserInfoVM(string studentEmail, int testSubmissionId, double percentualScore)
        {
            StudentEmail = studentEmail;
            TestSubmissionId = testSubmissionId;
            PercentualScore = percentualScore;
        }

        /// <summary>
        /// email of the student
        /// </summary>
        public string StudentEmail { get; set; }

        /// <summary>
        /// id of the test submission
        /// </summary>
        public int TestSubmissionId { get; set; }

        /// <summary>
        /// percentual score gained from the test
        /// </summary>
        public double PercentualScore { get; set; }
    }
}